sycl/piapi/include/pi/plugin.hpp

```diff
@@ -41,14 +45,14 @@ class plugin {
 
   ~plugin() = default;
 
-  const RT::PiPlugin &getPiPlugin() const { return MPlugin; }
-  RT::PiPlugin &getPiPlugin() { return MPlugin; }
+  const pi::PiPlugin &getPiPlugin() const { return MPlugin; }
+  pi::PiPlugin &getPiPlugin() { return MPlugin; }
 
   /// Checks return value from PI calls.
   ///
   /// \throw Exception if pi_result is not a PI_SUCCESS.
-  template <typename Exception = cl::sycl::runtime_error>
-  void checkPiResult(RT::PiResult pi_result) const {
+  template <typename Exception = std::runtime_error>
```

Comment:
Why std? I think we should throw sycl exceptions.

Reply:
Again, this is because of the need to make PI independent of SYCL. However, that's not always possible, so I've included the DPC++ integration macro, so I can change it to default to `cl::sycl::runtime_error` when that macro is enabled.

Reply:
By the SYCL specification we have to throw exceptions of sycl type, not std ones. As a solution this function can be moved out of the PI library files to the SYCL RT level.